tests/PHPStan/Analyser/nsrt/mixed-subtract.php

```diff
@@ -0,0 +1,56 @@
+<?php // lint >= 8.0
+
+namespace SubtractMixed;
+
+use function PHPStan\Testing\assertType;
+
+/**
+ * @param int|0.0|''|'0'|array{}|false|null $moreThenFalsy
+ */
+function subtract(mixed $m, $moreThenFalsy) {
+	if ($m !== true) {
+		assertType("mixed~true", $m);
+		assertType('bool', (bool) $m); // mixed could still contain something truthy
+	}
+	if ($m !== false) {
+		assertType("mixed~false", $m);
+		assertType('bool', (bool) $m); // mixed could still contain something falsy
+	}
+	if (!is_bool($m)) {
+		assertType('mixed~bool', $m);
+		assertType('bool', (bool) $m);
+	}
+	if (!is_array($m)) {
+		assertType('mixed~array', $m);
+		assertType('bool', (bool) $m);
+	}
+
+	if ($m) {
+		assertType("mixed~0|0.0|''|'0'|array{}|false|null", $m);
+		assertType('true', (bool) $m);
+	}
+	if (!$m) {
+		assertType("0|0.0|''|'0'|array{}|false|null", $m);
+		assertType('false', (bool) $m);
+	}
+	if (!$m) {
+		if (!is_int($m)) {
+			assertType("0.0|''|'0'|array{}|false|null", $m);
+			assertType('false', (bool)$m);
+		}
+		if (!is_bool($m)) {
+			assertType("0|0.0|''|'0'|array{}|null", $m);
+			assertType('false', (bool)$m);
+		}
+	}
+
+	if (!$m || is_int($m)) {
+		assertType("0.0|''|'0'|array{}|int|false|null", $m);
+		assertType('bool', (bool) $m);
+	}
+
+	if ($m !== $moreThenFalsy) {
+		assertType('mixed', $m);
```

Comment:
seems the substraction logic atm cannot deal with "subtract more then falsy", as the type stays `mixed`.